### PR TITLE
update to new registerRelatedFilesProvider API

### PR DESCRIPTION
**Pull request by genlu**

### CODE REVIEW

File: src/lsptoolshost/copilot.ts [hunk 1]
```diff
@@ -35,58 +38,60 @@ export async function registerCopilotExtensionAsync(
...
-    await ext.activate();
-    const relatedAPI = ext.exports as CopilotRelatedFilesProviderRegistration | undefined;
-    if (!relatedAPI) {
-        if (isTraceLogLevel) {
-            channel.appendLine(
-                'Incompatible GitHub Copilot extension installed. Skip registeration of C# related files provider.'
-            );
+    ext.activate().then(() => {
+        const relatedAPI = ext.exports as CopilotRelatedFilesProviderRegistration | undefined;
+        if (!relatedAPI) {
```
Comment (dibarbet) [inline src/lsptoolshost/copilot.ts:43]: Seems like this should work, but I'm not super confident on how it works in JS.  Can you verify that nothing throws if you have an older version of copilot with the previous version of the API?

File: src/lsptoolshost/copilot.ts [hunk 2]
```diff
@@ -35,58 +38,60 @@ export async function registerCopilotExtensionAsync(
...
+            return;
         }
-        return;
-    }
 
-    if (isTraceLogLevel) {
-        channel.appendLine('registeration of C# related files provider for GitHub Copilot extension succeeded.');
-    }
+        if (isTraceLogLevel) {
+            channel.appendLine('registeration of C# related files provider for GitHub Copilot extension succeeded.');
```
Comment (dibarbet) [inline src/lsptoolshost/copilot.ts:53]:
```suggestion
            channel.appendLine('registration of C# related files provider for GitHub Copilot extension succeeded.');
```

**Review by dibarbet — APPROVED** [review]
lgtm assuming this gracesfully handles the previous version of the API

File: src/lsptoolshost/copilot.ts (hunk 1 above)
Comment (genlu) [reply src/lsptoolshost/copilot.ts:43]: Hmm, it seems to work with old API as well,. My TS knowledge is poor so this is unexpected to me, I thought `relatedAPI ` for copilot extension with old API would be undefined. What I saw under debugger is the registration just work and callback when called got an undefined cancellation token.

**Pull request merged**